File: content/snippets/python/design_patterns/object_adapter.py
class Adaptee:
    def specific(self): print('Adaptee specific')
class Target:
    def request(self): raise NotImplementedError
class ObjectAdapter(Target):
    def __init__(self): self.a = Adaptee()
    def request(self): self.a.specific()

if __name__ == '__main__':
    ObjectAdapter().request()

File: content/snippets/go/design_patterns/object_adapter.go
package main

import "fmt"

type Adaptee struct{}

func (Adaptee) Specific() { fmt.Println("Adaptee specific") }

type Target interface{ Request() }

type ObjectAdapter struct{ a Adaptee }

func (o ObjectAdapter) Request() { o.a.Specific() }
func main()                      { var t Target = ObjectAdapter{}; t.Request() }
